src/systemTest/java/com/flightstats/hub/client/CallbackResourceClient.java

```diff
@@ -0,0 +1,11 @@
+package com.flightstats.hub.client;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Url;
+
+public interface CallbackResourceClient {
+
+    @GET
+    Call<String> get(@Url String hostUrl);
```

Comment:
Is there a reason this is so generic compared to the hub clients? It seems like creating a retrofit client with the callback server base uri would let you make this more explicit, with a path of `\{webhookName}`. That would make the test below read a little more cleanly, and then you could specify that it's going to be a `List<String>` which you wouldn't have to parse manually...

Reply:
I need to know the callback server url to create a retrofit client. callback instance gets injected here .
Also response parsing limitation is on retrofit side. Idid not see any special converters for java data structure  type interestingly.
        return new Retrofit.Builder()
                    .baseUrl(hubBaseUrl)
                    .addConverterFactory(ScalarsConverterFactory.create())
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(new OkHttpClient.Builder().build())
                    .build();

Reply:
I'm not sure I follow. Would it be possible to start the server and then instantiate the retrofit client, since at that point you'd have the url?

Reply:
By special converter for Java data structures, do you mean that it can't parse a `List<String>` but it can parse a `Webhook`?